src/translator/quality_estimator.cpp

```diff
@@ -0,0 +1,294 @@
+#include "quality_estimator.h"
+
+namespace marian::bergamot {
+
+void UnsupervisedQualityEstimator::computeQualityScores(const Histories& histories, Response& response) const {
+  size_t sentenceIndex = 0;
+
+  for (const auto& history : histories) {
+    const std::vector<float> logProbs = std::get<1>(history->top())->tracebackWordScores();
+    response.qualityScores.push_back(computeSentenceScores(logProbs, response.target, sentenceIndex));
+    ++sentenceIndex;
+  }
+}
+
+Response::SentenceQualityEstimate UnsupervisedQualityEstimator::computeSentenceScores(
+    const std::vector<float>& logProbs, const AnnotatedText& target, const size_t sentenceIdx) const {
+  const std::vector<SubwordRange> wordIndexes = mapWords(logProbs, target, sentenceIdx);
```

Comment:
`wordIndices`* should be nicer? I don't particularly mind `wordIndexes` though.

Reply:
Commit f5cc477